ouroboros-consensus/test-util/Test/Util/TestTx.hs

```diff
@@ -0,0 +1,49 @@
+module Test.Util.TestTx (
+    -- * Transactions
+    TestTxId
+  , TestTx (..)
+  ) where
+
+import           Data.Word (Word64)
+import           Test.QuickCheck (Arbitrary (..), oneof)
+
+{-------------------------------------------------------------------------------
+  Test infrastructure: test transaction
+-------------------------------------------------------------------------------}
+
+-- | A simple transaction identifier used in testing 'Mempool' support.
+newtype TestTxId = TestTxId Word64
+  deriving (Show, Eq, Ord)
+
+instance Arbitrary TestTxId where
+  arbitrary           = TestTxId <$> arbitrary
+  shrink (TestTxId w) = [TestTxId w' | w' <- shrink w]
+
+-- | A simple valid or invalid transaction used in testing 'Mempool' support.
+--
+-- Currently, in the tests for the basic 'Mempool' operations and properties,
+-- we don't really need to know much more about a transaction other than
+-- whether it is valid or invalid. At the moment, we're only testing properties
+-- surrounding a basic subset of 'Mempool' functionality: When attempting to
+-- add transactions to the 'Mempool', those which are valid are stored and
+-- those which are invalid are dropped.
+--
+-- TODO:
+-- This will most definitely need to be extended in order to test other
```

Comment:
Thinking out loud: we could use a simplified UTxO:
```haskell
data TestTx = TestTx
  { txDep :: !(Maybe TestTxId)
  , txId  :: !TextTxId
  }
```
For a `TestTx` to be valid, its `txDep` must be in the chain *and* cannot be the `txDep` of another `TestTx` in the chain.

Criteria: easy to generate with good coverage of valid + invalid, easy to shrink, easy to read/understand what kind of transaction we have (in test output).

Hmm, whether a `TestTx` is valid will depend on other transactions (which makes sense, as this is what we're testing). Idea: what if we just add this as an extra constructor to your `TestTx`, then we have: valid, invalid, and dependent (without the `Maybe` in `txDep`). It gives us the advantages of your simple representation + triggering more cases that require dependencies.

Reply:
Sounds like a good idea to me. I also like the suggestion of adding a third constructor, `DepTestTx` or `DependentTestTx`. Will definitely become necessary as I continue working on and adding tests for the `Mempool` functionality.